aip/general/0213.md

```diff
@@ -9,35 +9,66 @@ placement:
 
-However, there are also cases where common structures are valuable, especially
-where a concept is well-known and it is sufficiently clear that it will not
-change. Common protos serve this use case.
+There are two kinds of common protos:
 
-## Guidance
-
-The public representation of APIs **should** be self-contained, meaning that
-all protos used by the API originate in the same proto `package`, except for
-common protos, which **may** be used freely in any API.
+- Organization-specific common protos, covering organization-specific concepts
+  such as a "viewport" in Maps.
+- Global common protos which are not domain-specific, such as "timestamp" or
+  "postal address".
 
-APIs **must not** define a set of API-specific common protos which live outside
-of its versioning structure. This prevents independent movement of particular
-versions and also causes problems for client libraries in many languages that
-compile the proto messages into classes.
+Where it is safe to share a single representation across multiple APIs,
+common protos can make it easier for clients to interact with those APIs.
+Concept-specific client code can be written once, and messages can be used from
+the response of one API in the request of another without clunky copying, for
+example.
 
-APIs **should not** directly depend on protos defined in other APIs. Instead,
-they should copy and paste the applicable messages into their own API. When
-doing so, APIs **should** keep the field names and numbers the same.
+This benefit comes with significant restrictions and limitations, however,
+and should not be attempted lightly.
 
-## Existing common protos
+## Guidance
 
-The common protos, which public-facing protos for an API **may** safely import,
+- Organization-wide common proto packages **must** end with `.type`, e.g.
+  `google.geo.type` or `google.shopping.type`.
+- Organizations **must** consult the API design team before creating a new
+  organization-wide common proto package.
+- Organizations **must not** define generic protos in organization-specific
+  common protos, instead preferring global common protos.
+- Common protos **must not** be "moved" from an organization-specific common
+  proto package to a global common package or vice versa.
+- Fields **should not** be added to existing messages.
+- Values **should not** be added to existing enums.
+- Fields **must not** be removed from existing messages.
+- Values **must not** be removed from existing enums.
+- While documentation **may** be clarified, it **should not** change the set
+  of valid values for any given field or change the meanings of existing values.
+- New proto messages and enums **may** be added to common proto packages.
+  - API teams **should** be aware that it may take some time for any changes
+    to propagate through publication to client libraries and other surfaces,
+    and **should** allow sufficient time for this to happen before using the new
+    messages and enums in their APIs.
```

Comment:
This first "should" doesn't feel actionable. I might consider some wording that focuses on the action:

```suggestion
  - API teams  **should** allow sufficient time for propagation to clients before using the new
    messages and enums in their APIs. Fields may take some time for any changes
    to propagate through publication to client libraries and other surfaces.
```

Reply:
Works for me, thanks.